libcudacxx/include/cuda/__iterator/constant_iterator.h

```diff
@@ -129,32 +157,76 @@ class constant_iterator
+    if constexpr (::cuda::std::is_signed_v<_Index>)
+    {
+      _CCCL_ASSERT(__index_ + __n >= 0, "The index must be greater than or equal to 0");
+    }
     __index_ += static_cast<_Index>(__n);
     return *this;
   }
 
-  //! @brief Decrements the stored index by \p __n
+  //! @brief Creates a copy of a @c constant_iterator advanced by a given number of elements
+  //! @param __iter The @c constant_iterator to advance
+  //! @param __n The amount of elements to advance
+  [[nodiscard]] _CCCL_API friend constexpr constant_iterator
+  operator+(constant_iterator __iter, difference_type __n) noexcept
+  {
+    __iter += __n;
+    return __iter;
+  }
+
+  //! @brief Creates a copy of a @c constant_iterator advanced by a given number of elements
+  //! @param __n The amount of elements to advance
+  //! @param __iter The @c constant_iterator to advance
+  [[nodiscard]] _CCCL_API friend constexpr constant_iterator
+  operator+(difference_type __n, constant_iterator __iter) noexcept
+  {
+    __iter += __n;
+    return __iter;
+  }
+
+  //! @brief Decrements a @c constant_iterator by a given number of elements
   //! @param __n The amount of elements to decrement
   _CCCL_API constexpr constant_iterator& operator-=(difference_type __n) noexcept
   {
-    _CCCL_ASSERT(__index_ - __n >= 0, "The index must be greater than or equal to 0");
+    if constexpr (::cuda::std::is_signed_v<_Index>)
+    {
+      _CCCL_ASSERT(__index_ - __n >= 0, "The index must be greater than or equal to 0");
+    }
     __index_ -= static_cast<_Index>(__n);
     return *this;
   }
 
-  //! @brief Compares two \c constant_iterator for equality. Only compares the index in the sequence
+  //! @brief Creates a copy of a @c constant_iterator decremented by a given number of elements
+  //! @param __n The amount of elements to decrement
+  //! @param __iter The @c constant_iterator to decrement
+  [[nodiscard]] _CCCL_API friend constexpr constant_iterator
+  operator-(constant_iterator __iter, difference_type __n) noexcept
+  {
+    __iter -= __n;
+    return __iter;
+  }
+
+  //! @brief Returns the distance between two @c constant_iterator
+  [[nodiscard]] _CCCL_API friend constexpr difference_type
+  operator-(const constant_iterator& __lhs, const constant_iterator& __rhs) noexcept
+  {
+    return static_cast<difference_type>(__lhs.__index_) - static_cast<difference_type>(__rhs.__index_);
+  }
+
+  //! @brief Compares two @c constant_iterator for equality. Comparess the index in the sequence
   [[nodiscard]] _CCCL_API friend constexpr bool
   operator==(const constant_iterator& __lhs, const constant_iterator& __rhs) noexcept
   {
     return __lhs.__index_ == __rhs.__index_;
   }
+
 #if _CCCL_STD_VER <= 2017
-  //! @brief Compares two \c constant_iterator for inequality. Only compares the index in the sequence
+  //! @brief Compares two @c constant_iterator for inequality. Comparess the index in the sequence
```

Comment:
```suggestion
  //! @brief Compares two @c constant_iterator for inequality. Compares the index in the sequence
```
(many more occurrences)